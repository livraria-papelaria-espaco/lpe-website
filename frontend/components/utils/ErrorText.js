import { Alert } from '@material-ui/lab';
import PropTypes from 'prop-types';
import React from 'react';

const ErrorText = ({ error }) => {
  if (error && process.env.NODE_ENV !== 'production') console.error(error);
  return <Alert severity='warning'>Ocorreu um erro</Alert>;
};

ErrorText.propTypes = {
  error: PropTypes.string,
};

ErrorText.defaultProps = {
  error: undefined,
};

export default ErrorText;
