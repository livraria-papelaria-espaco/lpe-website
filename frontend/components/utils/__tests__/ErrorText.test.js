import React from 'react';
import { shallow } from 'enzyme';
import ErrorText from '../ErrorText';

describe('ErrorText Utility (components/utils/ErrorText)', () => {
  it('Renders without crashing and matches snapshot', () => {
    const wrapper = shallow(<ErrorText />);
    expect(wrapper).toMatchSnapshot();
  });
});
