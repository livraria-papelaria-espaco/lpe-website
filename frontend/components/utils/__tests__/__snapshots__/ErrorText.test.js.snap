// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorText Utility (components/utils/ErrorText) Renders without crashing and matches snapshot 1`] = `
<WithStyles(ForwardRef(Alert))
  severity="warning"
>
  Ocorreu um erro
</WithStyles(ForwardRef(Alert))>
`;
